### ISSUE TITLE
CRITICAL - Add information to the notification payload

### ISSUE THREAD

**Issue by franck-boullier** [unee-t/bz-database#115]
# The problem: 

In order to be able to manage the new and improved notification mechanism, we need to add the following information to the payload of the notification message:
- Current `Status` for the case
- Current `Resolution` for the case
- New value of the thing that has been changed in the case.

# The objects we need to alter:

## DB triggers:
- `ut_prepare_message_case_activity`
- `ut_prepare_message_case_assigned_updated`
- `ut_prepare_message_case_invited`
- `ut_prepare_message_new_case`
- `ut_prepare_message_new_comment`

## DB procedures:
- `lambda_notification_case_assignee_updated`
- `lambda_notification_case_invited`
- `lambda_notification_case_new`
- `lambda_notification_case_updated`
- `lambda_notification_message_new_comment`

**Comment by franck-boullier** [comment]
Done in the DEV/Staging and in the DEMO envo.
Waiting to check if this works as intended:
- Make sure all the Uilicious automated tests are working as they should
- Make sure that @nbiton has enough to implement the new and improved notification preferences. (see https://github.com/unee-t/frontend/issues/331 for more details)

**Comment by kaihendry** [comment]
I'm not sure how to test this. There is a dev regression this monday morning, not sure if it's related. cc @nbiton

**Comment by nbiton** [comment]
I don't think it can be related

**Comment by franck-boullier** [comment]
> I'm not sure how to test this

@Kaihendry: see my earlier comment:

> Make sure all the Uilicious automated tests are still working as they should
> Make sure that @nbiton has enough to implement the new and improved notification preferences. (see unee-t/frontend#331 for more details)

Not sure how I can be clearer on how to test this 🤔

**Comment by kaihendry** [comment]
[There is a regression.](https://github.com/unee-t/bz-database/issues/117) I don't know if the changes here contributed to it as it's not easy to bisect the change.

Perhaps before making a change, do a [backup](https://github.com/unee-t/bugzilla-customisation/blob/master/backup/backup.sh) and post a link to the SQL? This would allow Devops person like myself to roll back to previous state to confirm. 

I would of course snapshot the DB before rolling back.


I could use an RDS snapshot, but then I would need to guess when the changes were made and choose a time before that. 

tl;dr We need to co-ordinate to make sure these changes work.

**Comment by franck-boullier** [comment]
> We need to co-ordinate to make sure these changes work.

Sure!

@kaihendry 
- the link to the script that will revert back to before I did the change to the lambda objects:
https://github.com/unee-t/bz-database/blob/schema_v4.32.0_lambda_improvements/db%20scripts%20to%20add%20objects%20to%20the%20schema/Add_all_lambda_related_objects_v4.31.sql
- The link to the script who seems to have introduced a regression in the BZ DB schema:
https://github.com/unee-t/bz-database/blob/schema_v4.32.0_lambda_improvements/db%20scripts%20to%20add%20objects%20to%20the%20schema/Add_all_lambda_related_objects_v4.32.sql

I'm going to try to understand what's wrong with the upgrade I did in the DEV/Staging and DEMO envo. It has to be something inside https://github.com/unee-t/bz-database/blob/schema_v4.32.0_lambda_improvements/db%20scripts%20to%20add%20objects%20to%20the%20schema/Add_all_lambda_related_objects_v4.32.sql
...

@nbiton can you please tell me if the DEV/Staging and DEMO environment are generating the new payload correctly?

**Comment by kaihendry** [comment]
On dev the lambda call accountIDs appear incorrect. https://media.dev.unee-t.com/2019-03-05/accountID.mp4

This is a manual check (not automatic alerting) at present via Postman

**Comment by kaihendry** [comment]
IIUC we need to build some test payloads for the revised **notification_type** payloads. 

I have previously kept these: https://github.com/unee-t/lambda2sns/tree/master/tests/events

And now they need to be updated. I am actually not 100% sure how these should be done, since I am not 100% sure how to trigger each event in the UI.

**Comment by nbiton** [comment]
Confirmed to be resolved.
The new message payload attributes are present:
- "current_status"
- "current_severity"
- "current_resolution"

Thanks to @kaihendry for helping me retrieve the recent samples of message payloads.

**Comment by nbiton** [comment]
Also:
- "new_value"
- "old_value"

These two are used for determining when a case has been made resolved
